event_handler.go

```diff
@@ -201,100 +201,116 @@ func replacePermalinkWithText(ctx *IrcContext, text string) string {
 	return text + "\n> " + message.Text
 }
 
+func printMessage(ctx *IrcContext, message slack.Msg, prefix string) {
+	user := ctx.GetUserInfo(message.User)
+	name := ""
+	if user == nil {
+		if message.User != "" {
+			log.Warningf("Failed to get user info for %v %s", message.User, message.Username)
+			name = message.User
+		} else {
+			name = message.Username
+		}
+	} else {
+		name = user.Name
+	}
+	// get channel or other recipient (e.g. recipient of a direct message)
+	channame := resolveChannelName(ctx, message.Channel, message.ThreadTimestamp)
+
+	text := message.Text
+	for _, attachment := range message.Attachments {
+		text = joinText(text, attachment.Pretext, "\n")
+		text = joinText(text, attachment.Title, "\n")
+		if attachment.Text != "" {
+			text = joinText(text, attachment.Text, "\n")
+		} else {
+			text = joinText(text, attachment.Fallback, "\n")
+		}
+		text = joinText(text, attachment.ImageURL, "\n")
+	}
+	for _, file := range message.Files {
+		text = joinText(text, ctx.FileHandler.Download(file), " ")
+	}
+
+	log.Debugf("SLACK msg from %v (%v) on %v: %v",
+		message.User,
+		name,
+		message.Channel,
+		text,
+	)
+	if name == "" && text == "" {
+		log.Warningf("Empty username and message: %+v", message)
+		return
+	}
+	text = replacePermalinkWithText(ctx, text)
+	text = ctx.ExpandUserIds(text)
+	text = ExpandText(text)
+	text = joinText(prefix, text, " ")
```

Comment:
the prefix should be joined for every line in case of multi-line messages, or only the first line will show up, which can make hard to understand what was modified and how. At the same time I'm afraid of very long messages (like a copy-paste): once edited, they would throw a wall of text, which gets even worse if edited multiple times

Reply:
This code will print the `(edited)` prefix only for the first line, other lines in multi-line message will be printed but without the prefix. 

For long messages in general we might consider builing a feature for on-depand access: we could store long messages in a file and produce a url where you can read them (similar to file attachments). I would find this very useful for code snippets that people tend to paste in slack.